### PR TITLE
Make Snowflake host configurable in unit tests and honor the configuration in asserts

### PR DESCRIPTION
**Pull request by ikudjoi**
Fix for #77

### CODE REVIEW

File: Snowflake.Data.Tests/SFBaseTest.cs [hunk 1]
```diff
@@ -51,16 +48,18 @@ public void SFTestSetup()
...
-                    testConnectionConfig.warehouse);
-                this.testConfig = testConnectionConfig;
+                // Add host to connection string only if it is specified.
+                var host = testConnectionConfig.host ?? string.Empty;
+                if (host != string.Empty)
+                {
+                    host = $"host={host};";
+                }
+
+                connectionString = $"scheme=https;{host}port=443;" +
```
Comment (ikudjoi) [inline Snowflake.Data.Tests/SFBaseTest.cs]: Using string interpolation to format the connection string.

File: Snowflake.Data.Tests/SFBaseTest.cs [hunk 2]
```diff
@@ -80,6 +79,9 @@ public class TestConfig
         [JsonProperty(PropertyName = "SNOWFLAKE_TEST_ACCOUNT", NullValueHandling = NullValueHandling.Ignore)]
         internal string account { get; set; }
 
+        [JsonProperty(PropertyName = "SNOWFLAKE_TEST_HOST", NullValueHandling = NullValueHandling.Ignore)]
+        internal string host { get; set; }
```
Comment (ikudjoi) [inline Snowflake.Data.Tests/SFBaseTest.cs]: Added configurable host like the connector itself also provides.

File: Snowflake.Data.Tests/SFConnectionIT.cs [hunk 3]
```diff
@@ -59,7 +62,7 @@ public void TestConnectViaSecureString()
                 conn.ConnectionString = connectionStringWithoutPassword;
                 conn.Password = password;
                 conn.Open();
-                Assert.AreEqual("TESTDB_DOTNET", conn.Database);
+                Assert.AreEqual(testConfig.database, conn.Database);
```
Comment (ikudjoi) [inline Snowflake.Data.Tests/SFConnectionIT.cs]: The database is configurable but here fixed database name was assumed. Check against configs instead.

File: Snowflake.Data.Tests/SFConnectionIT.cs [hunk 4]
```diff
@@ -164,12 +167,21 @@ public void TestConnectWithoutHost()
...
-                String connStrFmt = "account={0};user={1};password={2}";
-                conn.ConnectionString = String.Format(connStrFmt, testConfig.account,
+                string connStrFmt = "account={0};user={1};password={2}";
+                conn.ConnectionString = string.Format(connStrFmt, testConfig.account,
                     testConfig.user, testConfig.password);
-                conn.Open();
-                Assert.AreEqual(conn.State, ConnectionState.Open);
-                conn.Close();
+                // Check that connection succeeds if host is not specified in test configs, i.e. default should work.
+                if (string.IsNullOrEmpty(testConfig.host))
```
Comment (ikudjoi) [inline Snowflake.Data.Tests/SFConnectionIT.cs]: Connection attempt without specified host will not work if the AWS region is other than us-west-1. Added condition to the assert.

File: Snowflake.Data.Tests/SFConnectionIT.cs [hunk 5]
```diff
@@ -196,7 +213,8 @@ public void TestConnectWithDifferentRole()
                     Assert.AreEqual(command.ExecuteScalar().ToString(), "INFORMATION_SCHEMA");
 
                     command.CommandText = "select current_warehouse()";
-                    Assert.AreEqual(command.ExecuteScalar().ToString(), "SHIGE_WH");
+                    // Command will return empty string if the hardcoded warehouse does not exist.
+                    CollectionAssert.Contains(new [] { "SHIGE_WH", "" }, command.ExecuteScalar().ToString());
```
Comment (ikudjoi) [inline Snowflake.Data.Tests/SFConnectionIT.cs]: Here it is assumed that a fixed, non-configurable warehouse exists. Added workaround.

**Comment by ikudjoi** [comment]
Hi, you never commented anything on this fine PR. Would make development of the connector easier for externals.

I'd also like to ask when you plan to release 1.0.6 with the overflow fix?

File: Snowflake.Data.Tests/SFConnectionIT.cs (hunk 4 above)
Comment (howryu) [reply Snowflake.Data.Tests/SFConnectionIT.cs]: I would like to make host a required test config.

**Comment by howryu** [comment]
Merged in #118

**Pull request closed without merging**